### PR TITLE
[#485] Writeable properties metaclass and loadable xml parser setting

### PR DESCRIPTION
**Pull request by d-w-moore**
This PR supplies the infrastructure to load a choice for (default) XML parser to be used in a PRC interpreter instance.
Some tricks of the Python trade are necessary, like allowing settable properties to alias to an already established variable state that determines a pre-existing PRC behavior. (Such was the case for the XML parser.)

### CODE REVIEW

**Comment by d-w-moore** [comment]
Taking out of draft status for review.

File: README.rst
Comment (d-w-moore) [reply README.rst]: Table replaced!

File: irods/client_configuration/__init__.py
Comment (d-w-moore) [reply irods/client_configuration/__init__.py]: Done and done.

File: irods/message/__init__.py
Comment (d-w-moore) [reply irods/message/__init__.py]: Ditto....

File: README.rst
Comment (trel) [inline README.rst:339]: nonzero -> non-zero ?

File: irods/message/__init__.py
Comment (trel) [inline irods/message/__init__.py:97]: let's remove dwm - okay to have a debug comment, but let's document as such.

**Review by alanking — COMMENTED** [review]
Doc changes look good. I don't see the changes for the `.py` files, though. Maybe forgot to add?

**Comment by trel** [comment]
@alanking i see three files changed, two are `__init__.py`.

**Comment by alanking** [comment]
Oh! It didn't show me that for some reason. Grr...

File: README.rst
Comment (d-w-moore) [reply README.rst]: Ok - corrected.

File: irods/message/__init__.py
Comment (d-w-moore) [reply irods/message/__init__.py]: Comment is removed.

**Review by alanking — COMMENTED** [review]
Looks good to me. Go ahead and squash (omit #s).

**Comment by d-w-moore** [comment]
Squashed.  Added issue just now for override environment variables for configuration settings.

**Review by alanking — DISMISSED** [review]
I think this is fine. We should probably add some tests if there aren't any already (sorry, can't remember). That can happen in another PR.

Let's wait for another approval before we put it in.

**Comment by d-w-moore** [comment]
> I think this is fine. We should probably add some tests if there aren't any already (sorry, can't remember). That can happen in another PR.
> 
> Let's wait for another approval before we put it in.

Agree tests are needed, but probably better in another PR.

File: irods/client_configuration/__init__.py
Comment (trel) [inline irods/client_configuration/__init__.py:182]: link to the py3 issue https://github.com/irods/python-irodsclient/issues/480

Comment (d-w-moore) [reply irods/client_configuration/__init__.py]: Look good?

**Review by alanking — COMMENTED** [review]
Seems okay to me. thinking on it, It may be helpful to be more specific about what X and Y are in the "raise X from Y" comment (e.g. use "raise RuntimeError from loadexc" ...just guessing)

File: irods/client_configuration/__init__.py
Comment (d-w-moore) [reply irods/client_configuration/__init__.py]: I guess I'm lazy in that regard and assume people will know what I mean ; ).  Yes, will be more specific

Comment (d-w-moore) [reply irods/client_configuration/__init__.py]: So, I'm deleting the comment completely.  I've been on a 360-degree turn about this today, and decided finally the "raise from" syntax is not even appropriate here because we're not transforming one exception to another within an except clause (the Python version of a C++ "catch") , which was the main rationalization for the Python team in creating the  "raise from" mechanism in the first place... i.e. to avoid the confusing "During handling of the above exception, another exception occurred ...".  See [this](https://docs.python.org/3/tutorial/errors.html#exception-chaining)

I'm also reverting to the original logic now default behavior for PRC is not to raise the `RuntimeError` because it's fatal for the irods import, and that's not what a user typically wants.  Plus I've decided the logic is simpler and more correct if  opting-in for the RuntimeError to be generated should be a simple and predictable effect of the user's choice of value for the environment variable, `PYTHON_IRODSCLIENT_CONFIGURATION_LOAD_ERRORS_FATAL`

**Comment by d-w-moore** [comment]
> Seems okay to me. thinking on it, It may be helpful to be more specific about what X and Y are in the "raise X from Y" comment (e.g. use "raise RuntimeError from loadexc" ...just guessing)

See above section.  Removing the comment about "raise from" completely.

File: irods/client_configuration/__init__.py
Comment (d-w-moore) [reply irods/client_configuration/__init__.py]: We'll discuss at next standup but I  do believe things are now correct as they stand.

Comment (alanking) [reply irods/client_configuration/__init__.py]: Nice explanation. Agreed. Resolving.

**Review by alanking — APPROVED** [review]
Please add #s

**Pull request merged**
